frame/nomination-pools/src/migration.rs

```diff
@@ -52,9 +52,12 @@ pub mod v1 {
 
 	impl<T: Config> OldBondedPoolInner<T> {
 		fn migrate_to_v1(self) -> BondedPoolInner<T> {
+			// Note: `commission` field not introduced to `BondedPoolInner` until
+			// migration 4.
 			BondedPoolInner {
-				member_counter: self.member_counter,
 				points: self.points,
+				commission: Commission::default(),
```

Comment:
To fix this, we should store a version of the `BondedPoolInner` per migration module. So, there is `BondedPoolInnerV0` (legacy), `BondedPoolInnerV1` in this module (which will be the one with `roles` but without). Even in the latest migration module, we should store a copy of the existing `BondedPoolInner` called e.g. `BondedPoolInnerv2` and migrate to `BondedPoolInnerv2`. `#[frame_support::storage_alias]` can be handy here to create fake storage types.

Reply:
I acknowledge that this is beyond the scope of this PR, so feel free to ignore but please do make a note or issue for our project to improve this process.

Reply:
I posted an issue here: https://github.com/paritytech/substrate/issues/13024, referencing this & your migrations PR as a foundation.